### PR TITLE
Configure isort, rearrange imports and remove circular dependency

### PR DESCRIPTION
**Pull request by dhruvkb**
## Description

This PR
- fixes a broken `isort` configuration
- formats the codebase to prevent lint failures
- improves 1 inconsistent and 1 cyclical import
- introduces a ton of merge conflicts so please merge everything else first

## Testing Instructions

Passing CI is sufficient.

## Review Instructions
While a lot of files have changed, you can ideally skip the commit **"Run lint to fix import order as per updated isort"** i.e. 6ec1dfa to get a clearer idea of what I changed vs what `isort` did.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by dhruvkb** [comment]
I've marked this as blocked till all other PRs are merged (tough that it'll happen, but no other way around it).

**Review by AetherUnbound — APPROVED** [review]
I like these changes! I know this won't get in just yet, but I like the separations.

File: .isort.cfg [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[settings]
+profile=black
+
+sections=FUTURE,STDLIB,DJANGO,THIRDPARTY,FIRSTPARTY,LOCALFOLDER
+known_django=django,rest_framework
+known_first_party=catalog
```
Comment (sarayourfriend) [inline .isort.cfg:6]: Nice :grin:

**Review by sarayourfriend — APPROVED** [review]
Love the import separation! Something I miss from the frontend. Thanks for adding it here.

Changes LGTM!

**Pull request merged**